src/app/backend/handler/apihandler.go

```diff
@@ -173,13 +177,53 @@ func FormatResponseLog(resp *restful.Response, req *restful.Request) string {
 		req.Request.RemoteAddr, resp.StatusCode())
 }
 
-// CreateHTTPAPIHandler creates a new HTTP handler that handles all requests to the API of the backend.
-func CreateHTTPAPIHandler(client *clientK8s.Clientset, heapsterClient client.HeapsterClient,
-	clientConfig clientcmd.ClientConfig) (http.Handler, error) {
+func (apiHandler *APIHandler) injectApiClient(fn func(*clientK8s.Clientset, *restful.Request, *restful.Response)) restful.RouteFunction {
+	return func(request *restful.Request, response *restful.Response) {
+		apiclient, _, err := apiHandler.getApiClient(request)
+		if err != nil {
+			handleInternalError(response, err)
+			return
+		}
+
+		fn(apiclient, request, response)
+	}
+}
+
+func (apiHandler *APIHandler) getApiClient(request *restful.Request) (*clientK8s.Clientset, clientcmd.ClientConfig, error) {
+	authorizationHeader := request.HeaderParameter("Authorization")
+	token := ""
+	if strings.HasPrefix(authorizationHeader, "Bearer ") {
+		token = strings.TrimPrefix(authorizationHeader, "Bearer ")
+	}
+
+	clientConfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+		&clientcmd.ClientConfigLoadingRules{ ExplicitPath: apiHandler.clientConfig.KubeConfigFile},
+		&clientcmd.ConfigOverrides{
+			AuthInfo:clientcmdapi.AuthInfo{Token: token},
```

Comment:
Quoting from https://github.com/kubernetes/client-go/blob/master/tools/clientcmd/overrides.go#L29:

``` go
// ConfigOverrides holds values that should override whatever information is pulled from the actual Config object.  You can't
// simply use an actual Config object, because Configs hold maps, but overrides are restricted to "at most one"
type ConfigOverrides struct {
	AuthInfo clientcmdapi.AuthInfo
	// ClusterDefaults are applied before the configured cluster info is loaded.
	ClusterDefaults clientcmdapi.Cluster
	ClusterInfo     clientcmdapi.Cluster
	Context         clientcmdapi.Context
	CurrentContext  string
	Timeout         string
}
```

I'm not sure how to understand "at most one" in this context.